www-client/firefox: bump to 98.0.2

ArchFeh opened the pull request:
Compared to the original ebuild, a patch was added and the sandbox was disabled because with 'enable-sanbox' caused many failures. This feature is temporarily not supported on riscv. Some logs like:

/var/tmp/portage/www-client/firefox-98.0.2/work/firefox-98.0.2/security/sandbox/linux/reporter/SandboxReporter.cpp:35:4: error: #error "unrecognized architecture"
164:47.00 /var/tmp/portage/www-client/firefox-98.0.2/work/firefox-98.0.2/security/sandbox/chromium/sandbox/linux/system_headers/linux_signal.h:65:2: error: #error "Unsupported platform"
164:47.58 /var/tmp/portage/www-client/firefox-98.0.2/work/firefox-98.0.2/security/sandbox/chromium/sandbox/linux/bpf_dsl/seccomp_macros.h:350:2: error: #error Unsupported target platform

Signed-off-by: Yu Gu <guyu2876@gmail.com>

dlan17 commented on www-client/firefox/firefox-98.0.2.ebuild:
```diff
@@ -725,11 +726,11 @@ src_configure() {
-	else
-		mozconfig_add_options_ac '' --enable-sandbox
-	fi
+	#if use ppc64; then
```
it's ugly, can we just use "use ppc64 || use riscv"?

ArchFeh replied:
Yes it's better.

dlan17 commented:
please slightly improve the commit message
1) always prefix with ${cat}/${pkg}, consist with official tree
2) why need disable sandbox? any reason, or failure?

Commit:
www-client/firefox: import ebuild from gentoo's official tree

original version without modification here

Signed-off-by: Yu Gu <guyu2876@gmail.com>

Commit:
www-client/firefox: bump to 98.0.2

Compared to the original ebuild, a patch was added and the sandbox was disabled because with 'enable-sanbox' caused many failures. This feature is temporarily not supported on riscv. Some logs like:

/var/tmp/portage/www-client/firefox-98.0.2/work/firefox-98.0.2/security/sandbox/linux/reporter/SandboxReporter.cpp:35:4: error: #error "unrecognized architecture"
164:47.00 /var/tmp/portage/www-client/firefox-98.0.2/work/firefox-98.0.2/security/sandbox/chromium/sandbox/linux/system_headers/linux_signal.h:65:2: error: #error "Unsupported platform"
164:47.58 /var/tmp/portage/www-client/firefox-98.0.2/work/firefox-98.0.2/security/sandbox/chromium/sandbox/linux/bpf_dsl/seccomp_macros.h:350:2: error: #error Unsupported target platform

Signed-off-by: Yu Gu <guyu2876@gmail.com>
